FluentTerminal.SystemTray/Utilities.cs

```diff
@@ -569,44 +569,58 @@ internal static void SaveFile(string path, string content)
             }
         }
 
-        internal static void MuteTerminal(bool mute)
+        private static bool MuteConhost(bool mute, int attemptCount)
         {
-            bool? deviceMuted = VolumeControl.GetDefaultAudioEndpointMute();
-            if (deviceMuted != true)
-            {
-                VolumeControl.SetDefaultAudioEndpointMute(true);
-            }
-
-            Process cmdProcess = new Process();
-            cmdProcess.StartInfo.FileName = "cmd.exe";
-            cmdProcess.StartInfo.WindowStyle = ProcessWindowStyle.Hidden;
-            cmdProcess.StartInfo.Arguments = "/k \x07";
-            cmdProcess.Start();
-
-            bool wasConhostMuted = false;
+            bool audioSessionFound = false;
             int attempt = 0;
-            while (wasConhostMuted == false && attempt++ < 2 && !cmdProcess.HasExited)
+            while (audioSessionFound == false && attempt++ < attemptCount)
             {
-                cmdProcess.WaitForExit(1500);
-
                 foreach (Process conhost in Process.GetProcessesByName("conhost"))
                 {
                     try
                     {
-                        bool? isMuted = null;
-                        isMuted = VolumeControl.GetAudioSessionMute(conhost.Id);
-                        if (isMuted != null && isMuted != mute)
+                        bool? isAudioSessionMuted = null;
+                        isAudioSessionMuted = VolumeControl.GetAudioSessionMute(conhost.Id);
+                        if (isAudioSessionMuted != null && isAudioSessionMuted != mute)
                         {
                             VolumeControl.SetAudioSessionMute(conhost.Id, mute);
-                            wasConhostMuted = true;
                             Logger.Instance.Debug($"Mute state for conhost process with id={conhost.Id} was set to {mute}.");
                         }
+
+                        if (isAudioSessionMuted != null)
+                        {
+                            audioSessionFound = true;
+                        }
                     }
                     catch (Exception) { }

```

Comment:
Is hiding exceptions like this really a good idea? Can you be more specific about the kind of exceptions you're wanting to squash here? What about logging?

Reply:
Finally, I was not able to get any exception on 1809/1903 and just removed try/catch block